File: Horizon_v2_3/Source/horizon/Target.cpp
#include "Target.h"

namespace horizon {
Target::Target(XMLNode targetXMLNode) : 
	name(targetXMLNode.getAttribute("TargetName")),
	targetType(targetXMLNode.getAttribute("TargetType")),
	position(new Position(targetXMLNode.getChildNode("POSITION"))),
	value(0)
{	
	if(targetXMLNode.isAttributeSet("MaxTimes"))
		value = atoi(targetXMLNode.getAttribute("Value"));
	// USER - Add other Target Attributes Here
	/*minQualCM = atoi(targetNode.getAttribute("minQual_cm"));;
	freq_days = atoi(targetNode.getAttribute("freq_days"));;
	CC = targetNode.getAttribute("CC");;
	WX_Reg = targetNode.getAttribute("WX_Reg");;
	Comment = targetNode.getAttribute("Comment");;*/
}

Target::Target(const string& name, Position* position) : name(name), position(position), targetType(TARGET_TYPE_EARTH_LOCATION), value(0) {}

Target::Target(const string& name, Position* position, int value) : name(name), position(position), targetType(TARGET_TYPE_EARTH_LOCATION), value(value) {}

Position* Target::getPos() const {
	return position;
}
bool Target::operator==(const Target& other) const {
	return (name == other.name);
}
} // end namespace horizon

File: Horizon_v_3_0/Horizon_v_3_0/Target.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml;
using Horizon.Universe;

namespace Horizon
{
    /**
    * A named location denoting where a task is to be performed.
    * @author Einar Pehrson
    */
    class Target
    {
        /** The name of the target */
        String Name { get; set; }

        /** The type of the target */
        TargetType TargetType { get; set; }

        /** The position of the target */
        DynamicState DynamicState { get; set; }

        /** The value of the target */
        int Value { get; set; }

        // TODO: What should we do with these?
        /*
        int minQualCM;

        int freq_days;
        string CC;
        string WX_Reg;
        */

        String Comment { get; set; }

        public Target(String name, TargetType type, DynamicState dynamicState, int value)
        {
            this.Name = name;
            this.TargetType = type;
            this.DynamicState = dynamicState;
            this.Value = value;
        }

        /**
        * Creates a new target from the xmlNode data
        * @param targetNode the xmlNode which contains the relevant target information
        */
        public Target(XmlNode targetXmlNode)
        {

        }

    }

    public enum TargetType { FACILITY, LOCATION }
}
